lib/api_auth/request_drivers/rest_client.rb

```diff
@@ -28,6 +29,7 @@ def calculated_md5
         else
           body = ''
         end
+        @request.payload.inspect # FIXME hack to reset internal stream to position 0
```

Comment:
It looks like this was changed in this commit: https://github.com/mgomes/api_auth/commit/eb2005caeab030515c4bfc642d37571bde71ea5d

And you can see the final version on master here: https://github.com/mgomes/api_auth/blob/master/lib/api_auth/request_drivers/rest_client.rb#L27